### PR TITLE
Make links absolute to api.rubyonrails.org

### PR DESCRIPTION
**Pull request by albertoalmagro**
### Summary

Currently if someone was reading ActiveRecord and ActiveModel documentation from GitHub, _Learn more_ links were broken without giving any hint about how to continue reading. This PR makes links redirect to api.rubyonrails.org so that reading can be continued.

See #30306 for more information.

### CODE REVIEW

**Comment by albertoalmagro** [comment]
Travis failed with an error that seems to me unrelated to this PR https://travis-ci.org/rails/rails/jobs/266005722

**Comment by albertoalmagro** [comment]
I have update master and rebased. Travis is running again.

**Comment by lugray** [comment]
Can you also fix similarly broken links in `railties/RDOC_MAIN.rdoc`?

**Comment by albertoalmagro** [comment]
Sure @lugray !

**Comment by albertoalmagro** [comment]
@lugray those actually work, they are using `link:classes` instead of `link:files`, I wonder if this will work when browsing older Rails versions API pages.

**Comment by lugray** [comment]
From https://github.com/rails/rails/blob/master/railties/RDOC_MAIN.rdoc, the two README links 404.

**Comment by kaspth** [comment]
I thought these doc links were meant to be relative? cc @fxn

**Comment by chancancode** [comment]
Thanks for the pull request. The docs are not intended to be viewable from Github. We should not use absolute URLs because there are multiple places where these could be rendered from.

Examples:

* edgeapi.rubyonrails.org: http://edgeapi.rubyonrails.org/classes/ActiveRecord.html
* versioned docs: http://api.rubyonrails.org/v4.2.0/files/activerecord/README_rdoc.html
* etc..

And it will become outdated when/if we switch to HTTPS.

Basically the same kind of reasons you would use URL helpers instead of hardcoding URLs in your Rails app.

**Comment by albertoalmagro** [comment]
That's fine @chancancode , thanks!
